notebooks/WHISP_and_chains_the_easy_way.ipynb

```diff
@@ -1,104 +1,179 @@
 {
-  "nbformat": 4,
-  "nbformat_minor": 0,
-  "metadata": {
-    "colab": {
-      "provenance": []
-    },
-    "kernelspec": {
-      "name": "python3",
-      "display_name": "Python 3"
-    },
-    "language_info": {
-      "name": "python"
-    }
-  },
   "cells": [
     {
       "cell_type": "markdown",
-      "source": [
-        "# WHISP pure Cloud Function"
-      ],
       "metadata": {
         "id": "GxxaiImdAlDC"
-      }
+      },
+      "source": [
+        "# WHISP pure Cloud Function\n",
+        "\n",
+        "WHISP (WHat IS in that Plot?) is a tool developed by FAO to aid in due diligence reporting for supply chain sustainability.  This notebook demonstrates how to use [WHISP](https://openforis.org/solutions/whisp/) to get sustainable sourcing information from the Google geospatial stack to your workflow.\n",
+        "\n",
+        "**WARNING**: These demos consume billable resources and may result in charges to your account!"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "# Setup"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "## Import Python packages for Setup"
+      ]
     },
     {
       "cell_type": "code",
+      "execution_count": 2,
+      "metadata": {},
+      "outputs": [],
       "source": [
-        "# REPLACE WITH YOUR PROJECT!\n",
-        "PROJECT = 'your-project'\n",
-        "# Suggested compute region:\n",
-        "REGION = 'us-central1'"
-      ],
-      "metadata": {
-        "id": "enVaKOcn7j5J"
-      },
-      "execution_count": null,
-      "outputs": []
+        "import os"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "## Determine the GCP project name and compute region"
+      ]
     },
     {
       "cell_type": "code",
+      "execution_count": 3,
+      "metadata": {},
+      "outputs": [],
       "source": [
-        "!gcloud auth login --project {PROJECT} --billing-project {PROJECT} --update-adc"
-      ],
+        "def is_running_in_colab():\n",
```

Comment:
imports in python are bound to the current scope, so this does not work as you intend in colab, unfortunately.

Reply:
possibly could just have different options depending on whether it's run in colab or not and describe the difference so a reader can do it correctly if there isn't an easy one-size-fits-all approach